operator/autoscaler.go

```diff
@@ -293,7 +293,9 @@ func (as *AutoScaler) scaleUpOrDown(esIndices map[string]ESIndex, scalingHint Sc
 			}
 		}
 
-		newDesiredNodeReplicas := as.ensureUpperBoundNodeReplicas(scalingSpec, int32(math.Ceil(float64(currentTotalShards)/float64(currentShardToNodeRatio-1))))
+		// round down to the next non-fractioned shard-to-node ratio
+		newShardToNodeRatio := math.Ceil(float64(currentTotalShards) / math.Floor(currentShardToNodeRatio-0.00001))
```

Comment:
This seems a little hacky, what is the reason for this particular fraction? :)

Reply:
just getting the code to round down. Too much magic? Shall I extract this into a private method and add more tests?

Reply:
I added more tests... WDYT?